### ISSUE TITLE
install.sh installs old version that is incompatible with macosx

### ISSUE THREAD

**Issue by OrenjiAkira** [DhavalKapil/luaver#21]
might be related to #12, #11 

Both running `install.sh` and installing luaver with homebrew installs an old version of luaver. This basically makes the function `__luaver_get_platform()` outdated. In MacOSX systems this means that luaver thinks it's running on linux, and compilation obviously fails.

**Comment by OrenjiAkira** [comment]
related to #8

**Comment by umireon** [comment]
@OrenjiAkira Homebrew version won't not run `install.sh`.
Have you read caveats of `luaver`?
They will be shown on the installation and appears on `brew info luaver`.

**Comment by OrenjiAkira** [comment]
@umireon The problem is that `install.sh` downloads a `luaver` script file from a server, which is an outdated version of the file. It doesn't matter whether you use homebrew or not. Manually editting the `luaver` script file after installation and apply the fix from #8 solves the problem locally. All that is needed is to update the file hosted in the server.

**Comment by umireon** [comment]
Yes, `install.sh` has a problem with macOS since it installs 1.0.0 which is not compatible with macOS.
And, no, homebrew version has no problem with macOS because [it is patched](https://github.com/Homebrew/homebrew-core/blob/master/Formula/luaver.rb#L12).
Have you added `if which luaver > /dev/null; then . `which luaver`; fi` in `.bashrc` or `.bash_profile` as written in caveats?

```
$ brew info luaver
luaver: stable 1.0.0, HEAD
Manage and switch between versions of Lua, LuaJIT, and Luarocks
https://github.com/DhavalKapil/luaver
/usr/local/Cellar/luaver/1.0.0 (5 files, 31.7K) *
  Built from source on 2016-12-20 at 03:28:51
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/luaver.rb
==> Dependencies
Required: wget ✔
==> Caveats
Add the following at the end of the correct file yourself:
  if which luaver > /dev/null; then . `which luaver`; fi
```

**Comment by OrenjiAkira** [comment]
I just retested the homebrew version and it works fine. When I created the issue I recall that even the homebrew version had this problem. I may be mistaken because the date of the patch is earlier than the issue by a couple of weeks, meaning it should have been patched by the time I tested it. Or I just don't remember it well. I apologize for not being updated on the homebrew version's status.

The issue remains regarding `install.sh`, though.

**Comment by umireon** [comment]
@OrenjiAkira No problem. Homebrew version has a bit confusing installation.
I hope the next release of `luaver` will solve the issue with `install.sh` on macOS.

**Comment by jdonaldson** [comment]
just out of curiosity, why doesn't homebrew have the latest version here?  Why does it need to be patched?

**Comment by umireon** [comment]
@jdonaldson This is because the latest code does not have any version.
[Homebrew never accept things without a stable version.](https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md#basic-instructions)
I really cannot do anything about that.

**Comment by jdonaldson** [comment]
Check the details on formula revisions:
https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md

You can specify a revision of a version, given as a commit hash.  It seems applicable here.

**Comment by umireon** [comment]
Formula revision is not intended to include an arbitrary version of softwave into Homebrew but to [modify Formula itself](https://github.com/Homebrew/homebrew-core/pull/8274#issuecomment-269689478).
Anyhow, [unversioned things will never be accepted in Homebrew](https://github.com/Homebrew/homebrew-core/pull/8226).

Instead, [you can try HEAD build](http://apple.stackexchange.com/questions/78984/what-does-brew-head-mean).

**Comment by jdonaldson** [comment]
Revisions were a bit overloaded, and I was speaking from outdated information, my bad : https://github.com/Homebrew/brew/issues/38

I'm pretty sure there's still a way to do this via git commit and version number.  If I have time, I'll try it out.

**Comment by umireon** [comment]
Thank you for clarification, @jdonaldson 😃 

FYI, The purpose of bottle and formula revisions are to maintain the versions of Homebrew package information itself.
And git revisions are to keep the integrity with git tags (i.e. to ensure the tag is not changed).

Technically, specifying a commit hash as a version is supported in brew but that is an unusual situation for package managers.
Of course, this feature can be used in private repositories.

Finally, the only and easiest way to make the Formula up to date seems to be releasing a new stable version, by design of Homebrew.

**Comment by jdonaldson** [comment]
Here's a pull request for brew-core for luaver: https://github.com/Homebrew/homebrew-core/pull/8747

It cleans up the install a bit, and enables git-based version and patching via revisions. Let me know if you like it.

**Comment by jdonaldson** [comment]
One more quick note... once a new stable version is released, you can change these lines:

```yaml
  url "https://github.com/DhavalKapil/luaver.git",
      :revision => "61861f2a966645a663971cd4e26d2a6104d075f6"
  version "1.0.0-p1"
```

to this:

```yaml
  url "https://github.com/DhavalKapil/luaver.git",
      :tag => "1.0.1"
```

should make upgrading a bit easier

**Comment by umireon** [comment]
The install.sh problem can be resolved by #43.

**Comment by DhavalKapil** [comment]
Closing this now. Let me know if you still face this issue.
